src/java.base/share/classes/java/lang/runtime/ObjectMethods.java

```diff
@@ -340,27 +340,29 @@ public static Object bootstrap(MethodHandles.Lookup lookup, String methodName, T
         }
         List<MethodHandle> getterList = List.of(getters);
         MethodHandle handle;
-        switch (methodName) {
-            case "equals":
+        return switch (methodName) {
+            case "equals"   -> {
                 if (methodType != null && !methodType.equals(MethodType.methodType(boolean.class, recordClass, Object.class)))
                     throw new IllegalArgumentException("Bad method type: " + methodType);
                 handle = makeEquals(recordClass, getterList);
-                return methodType != null ? new ConstantCallSite(handle) : handle;
-            case "hashCode":
+                yield methodType != null ? new ConstantCallSite(handle) : handle;
+            }
+            case "hashCode" -> {
                 if (methodType != null && !methodType.equals(MethodType.methodType(int.class, recordClass)))
                     throw new IllegalArgumentException("Bad method type: " + methodType);
                 handle = makeHashCode(recordClass, getterList);
-                return methodType != null ? new ConstantCallSite(handle) : handle;
-            case "toString":
+                yield methodType != null ? new ConstantCallSite(handle) : handle;
+            }
+            case "toString" -> {
                 if (methodType != null && !methodType.equals(MethodType.methodType(String.class, recordClass)))
                     throw new IllegalArgumentException("Bad method type: " + methodType);
                 List<String> nameList = "".equals(names) ? List.of() : List.of(names.split(";"));
                 if (nameList.size() != getterList.size())
                     throw new IllegalArgumentException("Name list and accessor list do not match");
                 handle = makeToString(recordClass, getterList, nameList);
-                return methodType != null ? new ConstantCallSite(handle) : handle;
-            default:
-                throw new IllegalArgumentException(methodName);
-        }
+                yield methodType != null ? new ConstantCallSite(handle) : handle;
+            }
+            default -> throw new IllegalArgumentException(methodName);
+        };
```

Comment:
I thinki it's simpler to have something like that
```
  var handle = switch(methodName) {
    ...
  };
  return methodType != null ? new ConstantCallSite(handle) : handle;
```

Reply:
I think that looks better. I've made that change now. See a8706b0